onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.h

```diff
@@ -44,6 +44,10 @@ namespace Dml
         gsl::span<const uint32_t> GetSizes() const { return { m_sizes, m_sizes + m_bufferTensorDesc.DimensionCount }; }
         gsl::span<const uint32_t> GetStrides() const;
         void SetStrides(gsl::span<const uint32_t> strides);
+        void PermuteDimensions(const std::vector<uint32_t> dimensionMapping, const TensorAxis alignment);
+        void PermuteArray(const std::vector<uint32_t> dimensionMapping, const TensorAxis alignment);
```

Comment:
>PermuteArray [](http://example.com/codeflow?start=13&length=12)

Avoid exposing publicly (internal anonymous namespace helper) since calling it would leave the TensorDesc in an incomplete state.

Reply:
Oops meant to add `PermuteArray` under private members